Whitespaces in path break unit tests build

**Description**
 When building unit tests with build directory that has whitespaces in it, build will fail:


**Steps to reproduce**
```
cmake -B "build dir" -S executables/unitTest -DBUILD_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=Debug
cmake --build build\ dir -j
```

**Expected behavior**
 Everything builds fine.

**Actual behavior**

```
/usr/bin/ld: cannot find /path/to/openbsw/build: file format not recognized
/usr/bin/ld: cannot find dir/openbsw/libs/bsw/bsp/mock/libbspMock.a: No such file or directory
collect2: error: ld returned 1 exit status
gmake[2]: *** [openbsw/platforms/posix/bsp/socketCanTransceiver/test/gtest/CMakeFiles/socketCanTransceiverTest.dir/build.make:126: openbsw/platforms/posix/bsp/socketCanTransceiver/test/gtest/socketCanTransceiverTest] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:4434: openbsw/platforms/posix/bsp/socketCanTransceiver/test/gtest/CMakeFiles/socketCanTransceiverTest.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
```

**Environment**
Any

--- a/platforms/posix/bsp/socketCanTransceiver/test/gtest/CMakeLists.txt
+++ b/platforms/posix/bsp/socketCanTransceiver/test/gtest/CMakeLists.txt
@@ -5,7 +5,7 @@ add_executable(
 
 target_link_libraries(
     socketCanTransceiverTest
-    PRIVATE "-Wl,--whole-archive $<TARGET_FILE:bspMock> -Wl,--no-whole-archive"
+    PRIVATE "-Wl,--whole-archive \"$<TARGET_FILE:bspMock>\" -Wl,--no-whole-archive"
             socketCanTransceiver bspMock gtest_main)
 
 gtest_discover_tests(socketCanTransceiverTest

